imap/conversations.c

```diff
@@ -285,22 +285,52 @@ EXPORTED int conversations_num_folders(struct conversations_state *state)
     return strarray_size(state->folders);
 }
 
-EXPORTED const char* conversations_folder_name(struct conversations_state *state,
-                                               int foldernum)
+EXPORTED const char *conversations_folder_mboxname(const struct conversations_state *state,
+                                                   int foldernum)
 {
-    const char *folder = strarray_safenth(state->folders, foldernum);
-    if (state->folders_byname) {
-        return folder;
-    }
-    else {
-        static char fname[MAX_MAILBOX_NAME+1];
+    const char *val = strarray_nth(state->folders, foldernum);
+    if (state->folders_byname)
+        return val;
+
+    // make it writeable even though const, because we're fiddling cache
+    struct conversations_state *backdoor = (struct conversations_state *)state;
+    if (!backdoor->altnames) backdoor->altnames = strarray_new();
+    const char *res = strarray_nth(backdoor->altnames, foldernum);
+    if (!res) {
+        if (!val) return NULL;
```

Comment:
nit: could merge this line with the above as `if (!res && val)`

Reply:
yeah true - or we could just put a `if (!val) return NULL;` way up top, because we don't even need to look for res - it's not going to be possible to resolve!